README.md

```diff
@@ -817,6 +825,37 @@ NOTE: Causes reparsing of the current translation unit.
 
 Supported in filetypes: `c, cpp, objc, objcpp`
 
+### The `FixIt` subcommand
+
+Where available, attempts to make changes to the buffer to correct the
+diagnostic closest to the cursor position.
+
+Completers which provide diagnostics may also provide trivial modifications to
+the source in order to correct the diagnostic. Examples include syntax errors
+such as missing trailing semi-colons, spurious characters, or other errors which
+the semantic engine can deterministically suggest corrections.
+
+If no fix-it is available for the current line, or there is no diagnostic on the
+current line, this command has no effect on the current buffer. If any
+modifications are made, the number of changes made to the buffer is echo'd and
+the user may use the editor's undo command to revert.
+
+When a diagnostic is available, and g:ycm_echo_current_diagnostic is set to 1,
+then the text `" (FixIt)"` is appended to the echo'd diagnostic when the
+completer is able to add this indication. The text `" (FixIt available)"` is 
+also appended to the diagnostic text in the output of the `:YcmDiags` command 
+for any diagnostics with available fix-its (where the completer can provide this
+indication).
+
+NOTE: Causes reparsing of the current translation unit.
+
+NOTE: After applying a fix-it, the cursor must be moved to update diagnostics,
```

Comment:
>  for some reason I don't fully understand, the diagnostics only refresh after (I presume) a "OnCursorMoved" event.

That's because of Vim technical restrictions. We can't update the UI without an event that triggers us.